### PR TITLE
Add ingress.useExisting value

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR adds a value allowing users to indicate they want to manage their own ingress but still want Teleport configured in ingress mode.

Changelog: The `teleport-cluster` chart can now use existing ingresses instead of creating its own.

### CODE REVIEW

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 1]
```diff
@@ -1892,6 +1892,25 @@ Boolean value that specifies whether to generate a Kubernetes `Ingress` for the
...
 
+## `ingress.useExisting`
+
+| Type      | Default value | Required? |
+|-----------|---------------|-----------|
+| `boolean` | `false`       | No        |
+
+
+`ingress.useExisting` indicates to the chart that you are providing your own ingress.
+The chart will configure Teleport like it's running behind an ingress,but will not
```
Comment (marcoandredinis) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:1903]:
```suggestion
The chart will configure Teleport like it's running behind an ingress, but will not
```

**Review by marcoandredinis — APPROVED** [review]

File: examples/chart/teleport-cluster/values.yaml [hunk 2]
```diff
@@ -700,6 +700,10 @@ service:
 # See https://goteleport.com/docs/architecture/tls-routing/#working-with-layer-7-load-balancers-or-reverse-proxies-preview for details.
 ingress:
   enabled: false
+  # Use existing indicates to the chart that you are providing your own ingress.
```
Comment (webvictim) [inline examples/chart/teleport-cluster/values.yaml:703]:
```suggestion
  # useExisting indicates to the chart that you are providing your own ingress.
```

**Review by webvictim — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 3]
```diff
@@ -1892,6 +1892,25 @@ Boolean value that specifies whether to generate a Kubernetes `Ingress` for the
...
+
+
+`ingress.useExisting` indicates to the chart that you are providing your own ingress.
+The chart will configure Teleport like it's running behind an ingress, but will not
+create the ingress resource. You are responsible for creating and managing the ingress.
+
+`values.yaml` example:
+
+```yaml
+ingress:
```
Comment (tigrato) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:1909]: should we start renaming this to the gateway api now that ingress api is frozen for quite some time?

File: examples/chart/teleport-cluster/values.yaml [hunk 4]
```diff
@@ -700,6 +700,10 @@ service:
 # See https://goteleport.com/docs/architecture/tls-routing/#working-with-layer-7-load-balancers-or-reverse-proxies-preview for details.
 ingress:
   enabled: false
+  # useExisting indicates to the chart that you are providing your own ingress.
```
Comment (tigrato) [inline examples/chart/teleport-cluster/values.yaml:703]:
```suggestion
  # useExisting indicates to the chart that you are managing your own ingress.
```

providing gives the impression that you need to provide the ingress to the chart

**Review by tigrato — APPROVED** [review]

File: examples/chart/teleport-cluster/values.yaml (hunk 4 above)
Comment (yair-segal) [reply examples/chart/teleport-cluster/values.yaml:703]: Hi @tigrato , same comment is probably valid also at docs/pages/reference/helm-reference/teleport-cluster.mdx at line 1902 in this branch.

File: docs/pages/reference/helm-reference/teleport-cluster.mdx (hunk 3 above)
Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-cluster.mdx:1909]: Last time I checked the gateway API is optional, its CRDs are not here by default. This makes the adoption for generic tools like us very hard.

I can extend the `ingress.useExisting` documentation that users can manage their own ingress, or HTTProute if they want. Using TCPRoute is roughly equivalent to using a ServiceLB and we must not allow self-signed+replication in this setup. TLSRoute experience will depend on the configuration. I don't think we can just replace "ingress" by "gateway API" as the new gwapi is way more powerful and allows user to shoot themselves in the foot in brand new ways.

Users are also thinking about ingress when they need to expose a service (and they also sadly don't consider L4 LoadBalancer Services, this just makes Teleport slower and more complex to deploy). I think ingress is the name everyone is looking for and we should not rename this section.

**Pull request merged**